File: src/C#/Parcial2/ActividadP2A4/EmpleadoBaseMasComision.cs
using System;

public class EmpleadoBaseMasComision : EmpleadoPorComision
{
    private double salarioBase;

    public EmpleadoBaseMasComision(string nombre, string apellido, string nss, double ventas, double tarifa, double salario)
        : base(nombre, apellido, nss, ventas, tarifa)
    {
        EstablecerSalarioBase(salario);
    }

    public void EstablecerSalarioBase(double salario) { salarioBase = (salario < 0.0) ? 0.0 : salario; }
    public double ObtenerSalarioBase() { return salarioBase; }

    public override double Ingresos() { return ObtenerSalarioBase() + base.Ingresos(); }

    public override string ToString()
    {
        return "con salario base " + base.ToString() + "\nsalario base: $" + ObtenerSalarioBase();
    }
}

File: src/Python/Parcial2/ActividadP2A4/empleadobasemascomision.py
from empleadoporcomision import EmpleadoPorComision

class EmpleadoBaseMasComision(EmpleadoPorComision):
    def __init__(self, nombre, apellido, nss, ventas, tarifa, salario):
        super().__init__(nombre, apellido, nss, ventas, tarifa)
        self.establecerSalarioBase(salario)

    def establecerSalarioBase(self, salario):
        if salario < 0.0:
            self.salarioBase = 0.0
        else:
            self.salarioBase = salario

    def obtenerSalarioBase(self):
        return self.salarioBase

    def ingresos(self):
        return self.obtenerSalarioBase() + super().ingresos()

    def __str__(self):
        return f"con salario base {super().__str__()}\nsalario base: ${self.obtenerSalarioBase()}"
